oauth2_http/java/com/google/auth/oauth2/InternalAwsSecurityCredentialsProvider.java

```diff
@@ -0,0 +1,262 @@
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.google.auth.oauth2;
+
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpContent;
+import com.google.api.client.http.HttpHeaders;
+import com.google.api.client.http.HttpMethods;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestFactory;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.json.GenericJson;
+import com.google.api.client.json.JsonParser;
+import com.google.appengine.repackaged.com.google.common.collect.ImmutableList;
+import com.google.auth.http.HttpTransportFactory;
+import com.google.common.annotations.VisibleForTesting;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+import javax.annotation.Nullable;
+
+class InternalAwsSecurityCredentialsProvider implements AwsSecurityCredentialsProvider {
+  private static final long serialVersionUID = 4438370785261365013L;
+
+  // Supported environment variables.
+  static final String AWS_REGION = "AWS_REGION";
+  static final String AWS_DEFAULT_REGION = "AWS_DEFAULT_REGION";
+  static final String AWS_ACCESS_KEY_ID = "AWS_ACCESS_KEY_ID";
+  static final String AWS_SECRET_ACCESS_KEY = "AWS_SECRET_ACCESS_KEY";
+  static final String AWS_SESSION_TOKEN = "AWS_SESSION_TOKEN";
+
+  static final String AWS_IMDSV2_SESSION_TOKEN_HEADER = "x-aws-ec2-metadata-token";
+  static final String AWS_IMDSV2_SESSION_TOKEN_TTL_HEADER = "x-aws-ec2-metadata-token-ttl-seconds";
+  static final String AWS_IMDSV2_SESSION_TOKEN_TTL = "300";
+
+  private final AwsCredentialSource awsCredentialSource;
+  private EnvironmentProvider environmentProvider;
+  private transient HttpTransportFactory transportFactory;
+  private final String region;
+
+  InternalAwsSecurityCredentialsProvider(
+      AwsCredentialSource awsCredentialSource,
+      EnvironmentProvider environmentProvider,
+      HttpTransportFactory transportFactory,
+      String region) {
+    this.environmentProvider = environmentProvider;
+    this.awsCredentialSource = awsCredentialSource;
+    this.transportFactory = transportFactory;
+    this.region = region;
+  }
+
+  public AwsSecurityCredentials getCredentials() throws IOException {
```

Comment:
IDE was making me have this public since I guess all methods from the interface are required to be public, assuming this is why we had the mismatch for the credential sources as well. Will fix it by making the parent abstract instead of an interface

Reply:
Changed to abstract class, and made all package-private